### PR TITLE
Add keyExists Java API

### PR DESCRIPTION
**Pull request by rhubner**
Add a new method to check if a key exists in the database. It avoids copying data between C++ and Java code.

### CODE REVIEW

File: java/src/main/java/org/rocksdb/RocksDB.java [hunk 1]
```diff
@@ -2417,6 +2417,92 @@ public List<ByteBufferGetStatus> multiGetByteBuffers(final ReadOptions readOptio
...
+  public boolean keyExists(
+      final ReadOptions readOptions, final byte[] key, final int offset, final int len) {
+    return keyExists(null, readOptions, key, offset, len);
+  }
+
+  public boolean keyExists(final ColumnFamilyHandle columnFamilyHandle,
+      final ReadOptions readOptions, final byte[] key) {
+    return keyExists(columnFamilyHandle, readOptions, key, 0, key.length);
+  }
+
```
Comment (alanpaxton) [inline java/src/main/java/org/rocksdb/RocksDB.java:2446]: My text for the comment would be this:

   * Check if a key exists in the database.
   * This method is not as lightweight as {@code keyMayExist} but it gives a 100% guarantee
   * of a correct result, whether the key exists or not.
   *
   * Internally it checks if the key may exist and then double checks with read operation
   * that confirms the key exists. This deals with the case where {@code keyMayExist} may return
   * a false positive.
   * 
   * The code crosses the Java/JNI boundary only once.

File: java/rocksjni/rocksjni.cc [hunk 2]
```diff
@@ -2214,6 +2214,108 @@ bool key_may_exist_direct_helper(JNIEnv* env, jlong jdb_handle,
   return exists;
 }
 
+jboolean key_exist_helper(JNIEnv* env, jlong jdb_handle, jlong jcf_handle,
```
Comment (alanpaxton) [inline java/rocksjni/rocksjni.cc:2217]: This should now be key_exists_helper to be consistent.

**Review by alanpaxton — COMMENTED** [review]
Some trivial changes but basically LGTM

**Comment by akankshamahajan15** [comment]
@rhubner Can you update the branch?

**Comment by hx235** [comment]
> @rhubner Can you update the branch?

Or if you could, can you select "Allow edits from maintainers" as in [screenshots](https://www.google.com/search?sca_esv=574277537&rlz=1C5GCEM_enUS947US947&q=Allow+edits+from+maintainers+github&tbm=isch&source=lnms&sa=X&ved=2ahUKEwjL95KNrf6BAxUOMlkFHYz8DAgQ0pQJegQIChAB&biw=1608&bih=812&dpr=1.1)

**Comment by rhubner** [comment]
> @rhubner Can you update the branch?

Rebase to the latest.

**Pull request closed without merging**